### ISSUE TITLE
Question on LF influence on Snorkel predictions based on coverage

### ISSUE THREAD

**Issue by astarostap** [snorkel-team/snorkel#1434]
I've observed that the more coverage a rule has, the more influence it has over the Snorkel model predictions.

What would happen if you took all the LFs (assuming they're precise enough) and found their coverage. Then, gave them a vote weight proportional to their coverage?

I was thinking this could be another baseline to compare Snorkel against in addition to majority vote.

**Comment by sadikneipp** [comment]
Maybe weights proportional to `coverage*emp.accuracy`? 🤔
Assuming there is a tradeoff there and you want to be mindful of how good a LF is and how much it covers.

**Comment by ajratner** [comment]
Hi @astarostap,

It's hard to comment on this without any specific details or empirical numbers, but yes: if a labeling function (or anything, more generally) labels more data points, it will end up having a bigger effect on the model being trained on those data points.  I'm not sure what the rationale would be though for further up-weighting high coverage LFs though?

Either way, a general comment: there are a million ad hoc ways one could potentially think of weighting and combining the LF outputs, all of which might seem reasonable in specific contexts... and all of which induce non-trivial tradeoffs as @sadikneipp points out!  This could lead to a lot of spinning around in circles!  The big reason why we built the label model the way we did is that the weights mean something in a precise and well-formalized way: they are estimated conditional probabilities of a well-defined probabilistic graphical model, and we prove theoretical guarantees about the estimation (along with lots of empirical results too).
